File: python/convert-integer-to-the-sum-of-two-no-zero-integers.py
class Solution:
    def is_no_zero(a: int) -> bool:
        while a:
            if a % 10 == 0:
                return False
            a //= 10
        return True

    def getNoZeroIntegers(self, n: int) -> List[int]:
        for a in range(1, n):
            b = n - a
            if Solution.is_no_zero(a) and Solution.is_no_zero(b):
                return [a, b]
        return None

File: rust/convert-integer-to-the-sum-of-two-no-zero-integers.rs
impl Solution {
    fn is_not_zero_integer(mut a: i32) -> bool {
        while a != 0 {
            if a % 10 == 0 {
                return false;
            }
            a = a / 10;
        }
        return true;
    }

    pub fn get_no_zero_integers(n: i32) -> Vec<i32> {
        let mut output = Vec::new();
        for a in 1..n {
            let b = n - a;
            //println!("a: {}, b: {}", a, b);
            if Solution::is_not_zero_integer(a) && Solution::is_not_zero_integer(b) {
                output.push(a);
                output.push(b);
                break;
            }
        }
        return output
    }
}
